AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClient.swift

```diff
@@ -328,6 +271,75 @@ final public class AWSMobileClient: _AWSMobileClient {
             listener.1(userState, additionalInfo)
         }
     }
+
+    private func registerIfPresentHostedUI() throws {
+        guard self.federationProvider == .hostedUI else { return }
+
+        loadHostedUIScopesFromKeychain()
+        loadOAuthURIQueryParametersFromKeychain()
+
+        let infoDictionaryMobileClient = self.awsInfo.rootInfoDictionary["Auth"] as? [String: [String: Any]]
+        let infoDictionary: [String: Any]? = infoDictionaryMobileClient?["Default"]?["OAuth"] as? [String: Any]
+
+        let clientId = infoDictionary?["AppClientId"] as? String
+        let secret = infoDictionary?["AppClientSecret"] as? String
+        guard let webDomain = infoDictionary?["WebDomain"] as? String else {
+            throw AWSMobileClientError.invalidConfiguration(
+                message: "WebDomain is missing in the configuration for hosted UI")
+        }
+        let hostURL = "https://\(webDomain)"
+        if self.scopes == nil {
+            self.scopes = infoDictionary?["Scopes"] as? [String]
+        }
+
+        let signInRedirectURI = infoDictionary?["SignInRedirectURI"] as? String
+        let signInURI = infoDictionary?["SignInURI"] as? String
+        if self.signInURIQueryParameters == nil {
+            self.signInURIQueryParameters = infoDictionary?["SignInURIQueryParameters"] as? [String: String]
+        }
+
+        let signOutRedirectURI = infoDictionary?["SignOutRedirectURI"] as? String
+        let signOutURI = infoDictionary?["SignOutURI"] as? String
+        if self.signOutURIQueryParameters == nil {
+            self.signOutURIQueryParameters = infoDictionary?["SignOutURIQueryParameters"] as? [String: String]
+        }
+
+        let tokensURI = infoDictionary?["TokenURI"] as? String
+        if self.tokenURIQueryParameters == nil {
+            self.tokenURIQueryParameters = infoDictionary?["TokenURIQueryParameters"] as? [String: String]
+        }
+
+        if (clientId == nil || scopes == nil || signInRedirectURI == nil || signOutRedirectURI == nil) {
```

Comment:
nit: unwrap these values here instead of force-unwrapping them below

Reply:
Added in https://github.com/aws-amplify/aws-sdk-ios/pull/4549